xen/arch/arm/coproc/coproc.c

```diff
@@ -353,24 +351,27 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
 
     for ( i = 0; i < num_mmios; ++i )
     {
-        res = platform_get_resource(pdev, IORESOURCE_MEM, i);
-        coproc->mmios[i].base = devm_ioremap_resource(dev, res);
-        if ( IS_ERR(coproc->mmios[i].base) )
+        struct mmio *mmio = &coproc->mmios[i];
+        ret = dt_device_get_address(np, i, &mmio->addr, &mmio->size);
+        if ( ret )
         {
-            printk("Failed to remap IO for \"%s\"\n", dev_path(coproc->dev));
-            ret = PTR_ERR(coproc->mmios[i].base);
+            printk("Failed to get mmio index %d for \"%s\"\n",
+                   i, dev_path(coproc->dev));
             goto out;
         }
 
-        coproc->mmios[i].size = resource_size(res);
-        coproc->mmios[i].addr = resource_addr(res);
-        coproc->mmios[i].coproc = coproc;
+        mmio->base = ioremap_nocache(mmio->addr, mmio->size);
+        if ( IS_ERR(mmio->base) )
```

Comment:
ioremap_nocache returns NULL when it fails to remap. Why do you check for IS_ERR?